repos/system_upgrade/el7toel8/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -57,88 +45,99 @@ def anyhasprefix(value, prefixes):
     return False
 
 
-@aslist
-def get_system_users():
+def get_system_users_status():
     ''' Get a list of users from `/etc/passwd` '''
-    for p in pwd.getpwall():
-        yield User(
-            name=p.pw_name,
-            uid=p.pw_uid,
-            gid=p.pw_gid,
-            home=p.pw_dir
-        )
+
+    @aslist
+    def _get_system_users():
+        for p in pwd.getpwall():
+            yield User(
+                name=p.pw_name,
+                uid=p.pw_uid,
+                gid=p.pw_gid,
+                home=p.pw_dir
+            )
+
+    return UsersFacts(users=_get_system_users())
 
 
-@aslist
-def get_system_groups():
+def get_system_groups_status():
     ''' Get a list of groups from `/etc/groups` '''
-    for g in grp.getgrall():
-        yield Group(
-            name=g.gr_name,
-            gid=g.gr_gid,
-            members=g.gr_mem
-        )
 
+    @aslist
+    def _get_system_groups():
+        for g in grp.getgrall():
+            yield Group(
+                name=g.gr_name,
+                gid=g.gr_gid,
+                members=g.gr_mem
+            )
+
+    return GroupsFacts(groups=_get_system_groups())
 
-@aslist
-def get_active_kernel_modules(logger):
+
+def get_active_kernel_modules_status(logger):
```

Comment:
logger doesn't need to be passed anymore, one _can_ use now [leapp.libraries.stdlib.api.current_logger()](https://leapp.readthedocs.io/en/latest/pydoc/leapp.libraries.stdlib.html#leapp.libraries.stdlib.api.current_logger) to get it
Anyway that's an FYI because I don't see you having to change this now